### PR TITLE
Add a Playwright ui test

### PR DESCRIPTION
**Pull request by steven-the-qa**
## Changes proposed 
- Set up Playwright for UI testing
- Add a test for the newsletter form's email validation (negative scenario to avoid actually signing up)
- Upload test results as artifacts after they run in CI
- Add a way to manually run Playwright tests from the GitHub repo (for fun)

## Screenshots
Screenshot 2024-10-11 at 11 29 04 PM

## Note to reviewers
Feel free to reject if this was overkill or not wanted 😆 I thought I'd have a little fun either way.

### CODE REVIEW

**Comment by steven-the-qa** [comment]
@elliezub there's a CI workflow I introduced that seems like it needs approval. It's meant to run the Playwright tests when a PR is created and upload the results afterward.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  "scripts": {
+    "test": "npx playwright test --project=chromium"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@playwright/test": "^1.48.0",
+    "@types/node": "^22.7.5",
+    "http-server": "^14.1.1"
```
Comment (steven-the-qa) [inline package.json:15]: the `http-server` is needed so we can serve up the `index.html` file for running UI tests

File: .github/workflows/playwright.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      with:
+        name: html-report
+        path: playwright-report/
+        retention-days: 5
+    - uses: actions/upload-artifact@v4
+      if: ${{ !cancelled() }}
+      with:
+        name: traces
+        path: test-results/
+        retention-days: 5
```
Comment (steven-the-qa) [inline .github/workflows/playwright.yml:33]: If you have any questions about this, I'm an open book. But also, asking any AI chat will probably give you a good explanation of anything that looks unfamiliar here 😁

File: index.html [hunk 3]
```diff
@@ -36,7 +36,7 @@ <h2>Let me know about upcoming Tech Commute news!</h2>
                             </div>
                             <div class="optionalParent">
                                 <div class="clear foot">
-                                    <input type="submit" name="subscribe" id="mc-embedded-subscribe" class="button" value="Subscribe">
+                                    <input data-testid="subscribeButton" type="submit" name="subscribe" id="mc-embedded-subscribe" class="button" value="Subscribe">
                                     <p style="margin: 0px auto;"><a href="http://eepurl.com/iDN-uA" title="Mailchimp - email marketing made easy and fun"></a></p>
                                 </div>
                             </div>
```
Comment (steven-the-qa) [inline index.html:42]: Test ids are a best practice in the test automation world. There are other ways to locate DOM nodes, but this is the most stable.

Comment (steven-the-qa) [inline index.html:42]: The test ids are the preferred approach among most QA Automation professionals for their stability. That said, your project, your rules. I can remove them and use different locators if you like.

**Comment by jasonetorres** [comment]
great add @steven-the-qa , approved the test and merged!

**Comment by jasonetorres** [comment]
thanks @steven-the-qa enjoy your @holopin
https://holopin.io/collect/cm297v5hw18430cl1rosd4p5s
